src/main/kotlin/com/lykke/me/subscriber/AbstractMeListener.kt

```diff
@@ -0,0 +1,22 @@
+package com.lykke.me.subscriber
+
+import java.util.concurrent.CopyOnWriteArraySet
+
+abstract class AbstractMeListener<T>: MeListener<T> {
+
+    private val subscribers = CopyOnWriteArraySet<MeSubscriber<T>>()
+
+    override fun subscribe(subscriber: MeSubscriber<T>) {
+        subscribers.add(subscriber)
+    }
+
+    override fun unsubscribe(subscriber: MeSubscriber<T>) {
+        subscribers.remove(subscriber)
+    }
+
+    protected fun notifySubscribers(message: T) {
+        subscribers.forEach {
+            it.notify(message)
```

Comment:
suggest to invoke it.notify in try catch - to let other subscribers to be invoked in case one of subscribers throw exception

Reply:
fixed. Also moved this class to kotlin utils lib